redbot/cogs/economy/economy.py

```diff
@@ -278,24 +277,30 @@ def __init__(self, bot: Red):
         if top < 1:
             top = 10
         if await bank.is_global():
-            bank_sorted = sorted(await bank.get_global_accounts(),
-                                 key=lambda x: x.balance, reverse=True)
+            bank_sorted = sorted(await bank.get_global_accounts(ctx.bot),
+                                 key=lambda x: x[1].balance, reverse=True)
+            if not check_global:
+                if guild is not None:
+                    bank_sorted = [x for x in bank_sorted if guild.get_member(x[0].id)]
         else:
-            bank_sorted = sorted(await bank.get_guild_accounts(guild),
-                                 key=lambda x: x.balance, reverse=True)
+            if guild is None:
+                await ctx.send(_("Bank is per-guild so this command cannot be used in DMs!"))
+                return
+            bank_sorted = sorted(await bank.get_guild_accounts(guild, ctx.bot),
+                                 key=lambda x: x[1].balance, reverse=True)
         if len(bank_sorted) < top:
             top = len(bank_sorted)
         topten = bank_sorted[:top]
         highscore = ""
         place = 1
         for acc in topten:
-            dname = str(acc.name)
-            if len(dname) >= 23 - len(str(acc.balance)):
-                dname = dname[:(23 - len(str(acc.balance))) - 3]
+            dname = str(acc[0].display_name)
```

Comment:
This should be probably be `acc[0].name` (or include a comment as to why),  `display_name` is misleading to anyone reading the code later as this will be a user object any time the leaderboard is global, and the behavior of this on a user object is identical to name, even if there is a nickname available as nickname is not a thing for user objects

Reply:
changed this now